src/components/common/FilterMultiSelect/index.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useEffect, useState } from 'react';
+import { Box, Theme, Typography, useTheme } from '@mui/material';
+import { Button, MultiSelect } from '@terraware/web-components';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+import strings from 'src/strings';
+import { makeStyles } from '@mui/styles';
+
+interface StyleProps {
+  isMobile?: boolean;
+}
+
+const useStyles = makeStyles((theme: Theme) => ({
+  button: {
+    width: (props: StyleProps) => (props.isMobile ? '100%' : 'auto'),
+  },
+  multiSelectStyle: {
+    height: '100%',
+    width: '100%',
+  },
+}));
+
+type FilterMultiSelectProps<T> = {
+  filterLabel: string;
+  initialSelection: T[];
+  onCancel: (finalSelection: T[]) => void;
```

Comment:
do we need the finalSelection in onCancel?

Reply:
not necessarily. I'm ignoring it in the Inventory implementation. I guess I can't think of a case where one would want it, so I'll take it out.

Reply:
that might be good, typically cancel is a void behavior.. parent doesn't need to know of any changes.